### PR TITLE
Drl 7 : On Adding Contact to opportunity on the contact lookup field.

### PR DESCRIPTION
**Pull request by deeksha-suvarna**
**Changes :** 

API_Plus_New_OpportunityPage.cmp
API_Plus_New_OpportunityPageController.js
API_Plus_New_OpportunityPageHelper.js

### CODE REVIEW

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 1]
```diff
@@ -576,6 +576,28 @@
             }
         });
         $A.enqueueAction(action);
+    },
+    validateTriggerSheetListAPI: function(component,event)
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:580]: Is this an existing method or did you create it? The method name is not meaningful in our scenario

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 2]
```diff
@@ -576,6 +576,28 @@
...
+    {
+        var contactId=component.get("v.contactId");
+        var isValid=true;
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:583]: Are we using isValid anywhere, if not remove it

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 3]
```diff
@@ -576,6 +576,28 @@
             }
         });
         $A.enqueueAction(action);
+    },
+    validateTriggerSheetListAPI: function(component,event)
+    {
+        var contactId=component.get("v.contactId");
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:582]:
```suggestion
        var contactId = component.get("v.contactId");
```

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 4]
```diff
@@ -576,6 +576,28 @@
...
+        var isValid=true;
+        var message;
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:584]: I don't see 'message' variable being used anywhere. Remove it

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 5]
```diff
@@ -576,6 +576,28 @@
...
+        if (contactId=='' || contactId==undefined )
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:585]:
```suggestion
        if (contactId === '' || contactId === undefined) {
```

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 6]
```diff
@@ -576,6 +576,28 @@
...
+        	{
+                isValid=false;                
+                var toastEvent = $A.get("e.force:showToast");
+                toastEvent.setParams({
+                    "title":"Error!",
+                    "message":'Contact can not be blank',
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:591]: Message can be in custom label

**Review by csankit0108 — CHANGES_REQUESTED** [review]

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js [hunk 7]
```diff
@@ -576,6 +576,28 @@
...
+                    "type":"error"
+            });
+            toastEvent.fire();  
+        }
+        else {
+            event.preventDefault(); //Prevent default submit
+            var response=component.find("APIOptyCreateForm").submit();
```
Comment (csankit0108) [inline force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:598]:
```suggestion
            var response = component.find("APIOptyCreateForm").submit();
```

**Review by csankit0108 — CHANGES_REQUESTED** [review]

File: force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js (hunk 1 above)
Comment (deeksha-suvarna) [reply force-app/main/default/aura/API_Plus_New_OpportunityPage/API_Plus_New_OpportunityPageHelper.js:580]: Created a new method for validation for API Record type.

**Review by csankit0108 — APPROVED** [review]

**Pull request merged**